identity/src/keypair.rs

```diff
@@ -669,10 +660,42 @@ impl TryInto<rsa::PublicKey> for PublicKey {
     }
 }
 
+#[cfg(feature = "ed25519")]
+impl From<ed25519::PublicKey> for PublicKey {
+    fn from(key: ed25519::PublicKey) -> Self {
+        #[allow(deprecated)] // TODO: Remove when PublicKey::Ed25519 is made opaque
+        PublicKey::Ed25519(key)
+    }
+}
+
+#[cfg(feature = "secp256k1")]
+impl From<secp256k1::PublicKey> for PublicKey {
+    fn from(key: secp256k1::PublicKey) -> Self {
+        #[allow(deprecated)] // TODO: Remove when PublicKey::Secp256k1 is made opaque
+        PublicKey::Secp256k1(key)
+    }
+}
+
+#[cfg(feature = "ecdsa")]
+impl From<ecdsa::PublicKey> for PublicKey {
+    fn from(key: ecdsa::PublicKey) -> Self {
+        #[allow(deprecated)] // TODO: Remove when PublicKey::Ecdsa is made opaque
+        PublicKey::Ecdsa(key)
+    }
+}
+
+#[cfg(all(feature = "rsa", not(target_arch = "wasm32")))]
+impl From<rsa::PublicKey> for PublicKey {
+    fn from(key: rsa::PublicKey) -> Self {
+        #[allow(deprecated)] // TODO: Remove when PublicKey::Rsa is made opaque
+        PublicKey::Rsa(key)
+    }
+}
+
 #[cfg(test)]
 mod tests {
     use super::*;
-    use crate::PeerId;
+    use crate::peer_id::PeerId;
```

Comment:
Please keep the diff minimal.

Reply:
Clippy / rust-analyzer kept giving me errors on this until I added the module. Any idea on what would make it 100% happy other than doing this?

Reply:
I think the error I keep seeing must be related to whether the "PeerId" feature is enabled or not. I'll revert it.

Reply:
The rustc error I get is definitely linked to a lack of "PeerId" feature enabled for _this_ test.

Because this test *can* be run with "ed25519" feature enabled but without feature "PeerId" I would suggest either keeing the added "peer_id::" module path or adding the "PeerId" feature requirement for this test.

Recommend the first option. Adding features just for tests seems like a more complex way to accomplish the same thing?

Reply:
Interesting, is `PeerId` not feature-gated properly? We should not have the type if the feature is not active, regardless of which path we use to import it.

Reply:
You're right, there were  `#[cfg(feature = "peerid")]` annotations missing in two places. I can add them which should resolve this issue

Reply:
Confirmed that adding the 2 feature gates fixes the issue. The (3 out of 7) test now run with `--features=ed25519` and the rustc errors disappear